winlogtimeline/ui/export_timeline.py

```diff
@@ -0,0 +1,217 @@
+        self.cancel_button.grid(row=i, column=0, padx=padding, pady=padding, sticky='NESW')
+        self.export_button.grid(row=i, column=2, padx=padding, pady=padding, sticky='NESW')
+        i += 1
+
+        self.work_container.columnconfigure(0, weight=4)
+        self.work_container.grid(row=0, column=0, columnspan=5, rowspan=3, sticky='EW')
+
+        self.container.columnconfigure(0, weight=4)
+        self.container.pack(fill=BOTH)
+
+        self.delimiter_label.grid_forget()
+        self.delimiter.grid_forget()
+        self.export_button.config(state=DISABLED)
+
+    def display_settings(self, option):
+        """
+        Chooses what settings to display based on the choice made in the type dropdown.
+        :param option: the option that was selected in the type dropdown. Format = FILE_EXT <FILE_TYPE_NAME>
+        """
+
+        padding = 3
+        file_extension = option.split(" ")[0]
+
+        if file_extension == ".csv":
+            self.delimiter_label.grid(row=2, column=0, padx=padding, sticky='NES')
+            self.delimiter.grid(row=2, column=1, columnspan=1, padx=padding, pady=padding, sticky='NESW')
+            self.export_button.config(state=NORMAL)
+
+        elif file_extension == ".pdf":
+            self.delimiter_label.grid_forget()
+            self.delimiter.grid_forget()
+            self.export_button.config(state=NORMAL)
+
+    def export(self):
+        """
+        Starts the export process.
+        :return:
+        """
+        columns = {pair[1].cget("text"): pair[0].get() for pair in self.columns}
+
+        if self.file_path.get() == "":
+            messagebox.showerror("Error", "Please provide a valid filename for the exported file.")
+            return
+        elif '.' in self.file_path.get():
+            messagebox.showerror("Error",
+                                 "Please provide a valid filename without an extension.\nInstead of '{}' you can use '{}'".format(
+                                     self.file_path.get(), self.file_path.get().split(".")[0]))
+            return
+
+        type = self.type_string.get().split(" ")[0]
+        if type == ".csv":
+            self.export_csv(columns)
+            return
+        elif type == ".pdf":
+            self.export_pdf(columns)
+            return
+
+    def export_csv(self, columns):
+        """
+        Exports in csv format
+        :param columns:
+        :return:
+        """
+        records = self.current_project.get_all_logs()
+
+        delimiter = None
+        if self.delimiter_string.get() == "Comma":
+            delimiter = ","
+        elif self.delimiter_string.get() == "Space":
+            delimiter = " "
+        elif self.delimiter_string.get() == "Tab":
+            delimiter = "\t"
+
+        from csv import writer
+
+        with open(self.current_project.get_path() + '/{}.csv'.format(self.file_path.get()), 'w') as csvfile:
```

Comment:
Shouldn't the user have some input on where the file is exported to?

Reply:
I figured no. I believe it makes sense to have the file exported to the project directory - it's at least a good use of the repository. @carterv Thoughts?

Reply:
Can we set that as default but allow them to change locations if they desire?